feat(doc): Add the minimum requirements needed for the Exoscale Key in the documentation

mlec1 opened the pull request:
# Description
Add the minimum required permission for the Exoscale Key. 
I think that might be useful to other people who don't want to set unnecessary permissions. 

Please don't hesitate to correct if some of the `operation` are not needed or some are missing. 
Please don't hesitate to modify the policy if you have a way to make it more secure.

## Checklist
(For exoscale contributors)

* [x] Changelog updated (under *Unreleased* block)
* [ ] Integration testing

## Testing

<!--
Describe the tests you did
-->

Tested the permissions with `lego`. They are enough to get a certificate.

sauterp commented on README.md:
```diff
@@ -68,6 +68,31 @@ metadata:
+          "action": "deny"
+        },
+        {
+          "expression": "operationin['list-dns-domain-records','get-dns-domain-record','list-dns-domains','update-dns-domain-record','create-dns-domain-record','delete-dns-domain-record']",
```
```suggestion
          "expression": "operation in['list-dns-domain-records','get-dns-domain-record','list-dns-domains','update-dns-domain-record','create-dns-domain-record','delete-dns-domain-record']",
```

not your fault, this is a bug where some whitespace is removed when you export an IAM policy from the Portal. We're working on i.

sauterp replied:
Is there a reason you didn't include the `get-operation` in this list? Does the webhook work without it?
(`get-operation` is listed on line 16)

sauterp replied:
It looks like you added `update-dns-domain-record` in addition to what we recommend on line 16. Do we really need it?

sauterp replied:
After checking the source code I'm certain, that we don't need the `update-dns-domain-record` operation. However we do need the `get-operation`.

sauterp commented on README.md:
```diff
@@ -68,6 +68,31 @@ metadata:
 
+Your Exoscale Key should have at least the following `operation` for your domain: `list-dns-domain-records`, `get-dns-domain-record`, `list-dns-domains`, `update-dns-domain-record`, `create-dns-domain-record`, `delete-dns-domain-record`
+
+Here is an example of the minimum policy required for the Exoscale Key:
```
```suggestion
Here is an example of the minimal policy required for the IAM role:
```

sauterp commented:
Hi,
thank you for this contribution, we appreciate it!

If you could also add a line to the changelog, that would be great.

sauterp commented on README.md:
````diff
@@ -68,6 +68,31 @@ metadata:
 type: Opaque
 ```
 
+Your Exoscale Key should have at least the following `operation` for your domain: `list-dns-domain-records`, `get-dns-domain-record`, `list-dns-domains`, `update-dns-domain-record`, `create-dns-domain-record`, `delete-dns-domain-record`
````
```suggestion
The IAM role policy of your key should allow at least the following `operation`s for your domain: `list-dns-domain-records`, `get-dns-domain-record`, `list-dns-domains`, `update-dns-domain-record`, `create-dns-domain-record`, `delete-dns-domain-record`
```

kobajagi replied:
There is already a note about required permissions [here](https://github.com/exoscale/cert-manager-webhook-exoscale/blob/master/README.md?plain=1#L16).

We can remove the old note now, it referenced the old IAM key type anyway.

mlec1 commented:
Sorry I initially made this Pull Request because I didn't see the line 16. 

I have updated the doc based on your comments. 

Just for the `get-operation` operation, I tested just with `lego`  which was successful without it, but I was surprised too. I added it in the list of operation and remove the `update-dns-records`.

sauterp commented on README.md:
````diff
@@ -68,6 +66,31 @@ metadata:
 type: Opaque
 ```
 
+The IAM role policy of your key should allow at least the following `operation`s for your domain: `list-dns-domains`, `list-dns-domain-records`, `get-dns-domain-record`, `get-operation`, `create-dns-domain-record` and `delete-dns-domain-record`
````
thank you for the changes!

One last request: can you please remove the `get-operation` from the list(again). We have confirmed that it isn't needed anymore. Sorry for the back and forth.

mlec1 replied:
Done

sauterp commented on README.md:
```diff
@@ -68,6 +66,31 @@ metadata:
+          "action": "deny"
+        },
+        {
+          "expression": "operation in ['list-dns-domains', 'list-dns-domain-records', 'get-dns-domain-record', 'get-operation', 'create-dns-domain-record', 'delete-dns-domain-record']",
```
```suggestion
          "expression": "operation in ['list-dns-domains', 'list-dns-domain-records', 'get-dns-domain-record', 'create-dns-domain-record', 'delete-dns-domain-record']",
```

sauterp commented on README.md:
````diff
@@ -68,6 +66,31 @@ metadata:
 type: Opaque
 ```
 
+The IAM role policy of your key should allow at least the following `operation`s for your domain: `list-dns-domains`, `list-dns-domain-records`, `get-dns-domain-record`, `get-operation`, `create-dns-domain-record` and `delete-dns-domain-record`
````
```suggestion
The IAM role policy of your key should allow at least the following `operation`s for your domain: `list-dns-domains`, `list-dns-domain-records`, `get-dns-domain-record`, `create-dns-domain-record` and `delete-dns-domain-record`
```
